features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/utils/LambdaExecutor.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.utils
+
+import android.os.Handler
+import android.os.Looper
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.LinkedBlockingDeque
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
+
+internal class LambdaExecutor<T>(
```

Comment:
this executor is quite complex and a bit over - engineered for me. I am not sure what you are trying to achieve with it ? If you need some work to be done on the result why not adding this in the caller thread (Main Thread in your case) in the `onSuccess` callback ?

Reply:
Indeed, you're correct that it can be done in a simpler way using the threadpoolexecutor from `Base64Serializer`. I've refactored and removed the `LambdaExecutor` class